src/components/ec-currency-input/ec-currency-input.vue

```diff
@@ -110,166 +110,151 @@
-    },
-    isInvalid() {
-      return !!this.errorMessage;
-    },
-    currenciesItems() {
-      return this.currencies.map(currency => ({ text: currency, value: currency, id: currency }));
-    },
-    currenciesTooltipOptions() {
-      const { content, placement = 'top' } = this.disabledCurrenciesTooltip || {};
-      if (!content) {
-        return null;
-      }
-      return { content, placement };
-    },
-    currencyModel: {
-      get() {
-        return this.currenciesItems.find(item => item.value === this.modelValue.currency);
-      },
-      set(item) {
-        this.$emit('update:modelValue', { ...this.modelValue, currency: item.value });
-      },
-    },
-    amountModel: {
-      get() {
-        return this.modelValue.amount;
-      },
-      set(value) {
-        this.$emit('update:modelValue', { ...this.modelValue, amount: value });
-      },
-    },
+  currenciesAreLoading: {
+    type: Boolean,
+    default: false,
   },
-  methods: {
-    getPopoverStyle() {
-      if (this.$refs.popoverWidthReference) {
-        return {
-          width: `${this.$refs.popoverWidthReference.offsetWidth}px`,
-        };
-      }
-
-      return null;
-    },
-    onAmountChange(evt) {
-      this.$emit('change', evt);
-      this.$emit('amount-change', evt);
-    },
-    onCurrencyChange(evt) {
-      this.currenciesHasFocus = true;
-      this.$emit('change', evt);
-      this.$emit('currency-change', evt);
-    },
-    onFocusCurrency() {
-      this.currenciesHasFocus = true;
-      this.$emit('focus');
-    },
-    prefixedDataTest(dataTestSuffix) {
-      const dataTestPrefix = this.$attrs['data-test'];
-      if (dataTestPrefix) {
-        const dataTestPrefixes = [...dataTestPrefix.split(' '), 'ec-currency-input'];
-        return dataTestPrefixes.map(dataTest => `${dataTest}__${dataTestSuffix}`).join(' ');
-      }
-
-      return `ec-currency-input__${dataTestSuffix}`;
-    },
+  disabledCurrenciesTooltip: {
+    type: Object,
   },
+  isCurrenciesDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isAmountDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  amountPlaceholder: {
+    type: String,
+  },
+  searchCurrencyPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCurrenciesText: {
+    type: String,
+    default: 'No results found',
+  },
+});
+
+const emit = defineEmits(['update:modelValue', 'change', 'focus', 'amount-change', 'currency-change', 'open', 'after-open']);
+
+const uid = getUid();
+let currenciesHasFocus = false;
+const popoverOptions = {
+  autoSize: 'min',
 };
+const popoverWidthReference = ref(null);
+const attrs = useAttrs();
+
+const id = computed(() => `ec-currency-input-field-${uid}`);
+const errorId = computed(() => (isInvalid.value ? `ec-currency-input-field-${uid}` : null));
+const isInvalid = computed(() => !!props.errorMessage);
```

Comment:
This is a redundant computed and we could use props.errorMessage instead.

Reply:
I think this just might be for convenience as we also use this constant twice within the template. Happy to change them all to props.errorMessage if you think this makes more sense though.

Reply:
It could be used in the way the errorTooltipMessage variable is used, and that will be even more semantic because right in the place errorMessage is used, we could see what "is invalid". That approach with computed isInvalid variable is suitable for cases when there is some logic under the hood of the computed callback function. Here there is no one.